src/v/cloud_storage/segment_meta_cstore.cc

```diff
@@ -86,13 +86,42 @@ void increment_all(std::tuple<Args...>& tup) {
 }
 
 /// The iters tuple is a tuple of std::optional<iterator-type>. The method
-/// checks if the optoinal referenced by index is not none and dereferences.
+/// checks if the optional referenced by index is not none and dereferences.
 template<segment_meta_ix ix, class T, class... Args>
 void value_or(const std::tuple<Args...>& iters, T& res) {
     const auto& it = std::get<static_cast<size_t>(ix)>(iters);
     res = T(*it);
 }
 
+/// zip first_tuple with second_tuple, calls map_fn on each pair of elements,
+/// collect results in a tuple if map_fn produces a result
+template<typename Fn>
+auto tuple_map(Fn&& map_fn, auto&& first_tuple, auto&& second_tuple) {
```

Comment:
you might be able to use boost hana or some ranges stuff for this.

Reply:
waiting for clang15 to be able to use ranges, and boost hana i didn't consider it